iroh-automerge-repo/src/main.rs

```diff
@@ -55,10 +82,72 @@ async fn main() -> anyhow::Result<()> {
 
     let args = Args::parse();
 
-    let endpoint = iroh::Endpoint::builder().discovery_n0().bind().await?;
+    // Pull in the secret key from the environment variable if it exists
+    // This key is the public Node ID used to identify the node in the network
+    // If not provided, a random key will be generated, and a new Node ID will
+    // be assigned each time the app is started
+    let secret_key = match std::env::var("IROH_NODE_SECRET_KEY") {
+        Ok(key_hex) => match decode(&key_hex) {
+            Ok(key_bytes) => {
+                if key_bytes.len() == 32 {
+                    let mut bytes = [0u8; 32];
+                    bytes.copy_from_slice(&key_bytes);
+                    Some(iroh::SecretKey::from_bytes(&bytes))
+                } else {
+                    println!(
+                        "invalid IROH_NODE_SECRET_KEY provided: expected 32 bytes, got {}",
+                        key_bytes.len()
+                    );
+                    None
+                }
+            }
+            Err(_) => {
+                println!("invalid IROH_NODE_SECRET_KEY provided: not valid hex");
+                None
+            }
+        },
```

Comment:
Feel free to use `SecretKey::from_str` for parsing this here instead (it expects hex encoding)